modules/EnsEMBL/Web/Component/DataExport/Protein.pm

```diff
@@ -76,9 +76,10 @@ sub default_file_name {
   my $data_object = $self->hub->param('g') ? $self->hub->core_object('gene') : undef;
   if ($data_object) {
     $name .= '_';
-    my $stable_id = $data_object->stable_id;
-    my ($disp_id) = $data_object->display_xref;
-    $name .= $disp_id || $stable_id;
+    my $versioned_stable_id = $data_object->stable_id_version;
+    # Replace '.' with '_' to avoid file extention clashes 
+    $versioned_stable_id =~ s/\./_/g; 
+    $name .= $versioned_stable_id;
```

Comment:
maybe double check what data_object returns if there is no version stable id and probably it needs to default to just stable_id (same for transcript.pm below)

Reply:
i would leave the previous code as it is and add the following code:
```
    my $versioned_stable_id = $data_object->stable_id_version;
    # Replace '.' with '_' to avoid file extention clashes 
    $versioned_stable_id =~ s/\./_/g; 
     $name .= $versioned_stable_id  || $disp_id || $stable_id;

```

Reply:
It will return `undef` if there is no versioned stabled ID present. I have updated it to use `stable_id` by default.